src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final double DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= table.length * DEFAULT_LOAD_FACTOR) {
+            resize();
+        }
+        int i = getIndex(key);
```

Comment:
it is bad practice to use short names try to avoid it
```suggestion
        int index = getIndex(key);
```